api/src/main/java/org/ccci/idm/user/Group.java

```diff
@@ -1,62 +1,56 @@
 package org.ccci.idm.user;
 
-import com.google.common.base.Joiner;
-import com.google.common.base.MoreObjects;
-import com.google.common.base.Objects;
+import com.google.common.base.Function;
+import com.google.common.base.Predicates;
+import com.google.common.collect.FluentIterable;
 
 import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
 import javax.annotation.concurrent.Immutable;
-import java.io.Serializable;
 import java.util.Arrays;
+import java.util.List;
 
 @Immutable
-public final class Group implements Serializable {
+public final class Group extends Dn {
     private static final long serialVersionUID = 8588784014544957895L;
 
-    private final String[] path;
-
-    private final String name;
+    private static final String LEGACY_PATH_TYPE = "ou";
+    private static final String LEGACY_NAME_TYPE = "cn";
+    private static final Function<String, Component> LEGACY_PATH = new Function<String, Component>() {
+        @Nullable
+        @Override
+        public Component apply(@Nullable final String value) {
+            return value != null ? new Component(LEGACY_PATH_TYPE, value) : null;
+        }
+    };
 
+    /**
+     * @deprecated Since 0.3.0, use {@link Group#Group(List)} instead.
+     */
+    @Deprecated
     public Group(@Nonnull final String[] path, @Nonnull final String name) {
```

Comment:
the 1 minor thing that is not backwards compatible (but doesn't matter with any of our current usages) is support for path & name types other than `ou` and `cn` respectively. But we only use those types for our current groups.